### PR TITLE
avoid generating internal frames

### PR DESCRIPTION
**Pull request by fwbrasil**
This PR makes the `Frame` macro fail to compile in case the frame is in the `kyo` package and fixes the methods in the codebase that were missing the implicit.

### CODE REVIEW

File: kyo-bench/src/main/scala/kyo/bench/Bench.scala [hunk 1]
```diff
@@ -42,6 +42,8 @@ end Bench
 
 object Bench:
 
+    inline given kyo.kernel.Frame = kyo.kernel.Frame.internal
```
Comment (fwbrasil) [inline kyo-bench/src/main/scala/kyo/bench/Bench.scala:45]: There are some situations where an internal frame is necessary. `Frame.internal` is an `private[kyo]` API that allows generating a frame within the `kyo` package.

File: kyo-examples/jvm/src/main/scala/examples/ledger/Models.scala [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package kyo.examples.ledger
+package examples.ledger
```
Comment (fwbrasil) [inline kyo-examples/jvm/src/main/scala/examples/ledger/Models.scala:1]: Move the examples out of the `kyo` package so they can generate frames.

File: kyo-caliban/src/main/scala/kyo/Schema.scala [hunk 3]
```diff
@@ -17,6 +17,7 @@ given zioSchema[R, A: Flat, S](using ev: Schema[R, A], ev2: (A < S) <:< (A < (Ab
             ev.toType_(isInput, isSubscription)
 
         override def resolve(value: A < S): Step[R] =
+            given Frame = Frame.internal
```
Comment (fwbrasil) [inline kyo-caliban/src/main/scala/kyo/Schema.scala:20]: There are some situations where an internal frame is necessary. Frame.internal is an private[kyo] API that allows generating a frame within the kyo package.

**Review by hearnadam — APPROVED** [review]

File: kyo-prelude/shared/src/main/scala/kyo/kernel/Frame.scala [hunk 4]
```diff
@@ -49,15 +50,37 @@ object Frame:
...
+        if !internal then
+            findEnclosing { sym =>
+                val fileName = pos.sourceFile.name
+                sym.fullName.startsWith("kyo") && !allowKyoFileSuffixes.exists(fileName.endsWith)
+            }.foreach { sym =>
+                report.errorAndAbort(
+                    s"""Frame cannot be derived within the kyo package: ${sym.owner.fullName}
+                       |
+                       |To resolve this issue:
+                       |1. Propagate the Frame from user code using an implicit parameter.
```
Comment (hearnadam) [inline kyo-prelude/shared/src/main/scala/kyo/kernel/Frame.scala:75]: `implicit`?? What is this, Scala 2.13?

**Review by hearnadam — COMMENTED** [review]
Looks like there is an issue with kyo-sttp files under JS

**Pull request merged**